test/sim/misc/accuracy.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe("Accuracy", function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it(`should round half down when applying a modifier`, function () {
+		battle = common.createBattle([
+			[{species: 'Butterfree', ability: 'compoundeyes', moves: ['sleeppowder']}],
+			[{species: 'Beldum', moves: ['poltergeist']}],
+		]);
+
+		battle.onEvent('Accuracy', battle.format, function (accuracy) {
+			assert.equal(accuracy, 98, 'CompoundEyes Sleep Powder should be 98% accurate');
+		});
+
+		battle.makeChoices();
+
+		battle = common.createBattle([
+			[{species: 'Butterfree', ability: 'victorystar', moves: ['fireblast']}],
+			[{species: 'Regirock', moves: ['sleeptalk']}],
+		]);
+
+		battle.onEvent('Accuracy', battle.format, function (accuracy) {
+			assert.equal(accuracy, 94, 'Victory Star Fire Blast should be 93% accurate');
+		});
+
+		battle.makeChoices();
+
+		battle = common.createBattle([
+			[{species: 'Butterfree', item: 'widelens', moves: ['fireblast']}],
+			[{species: 'Regirock', moves: ['sleeptalk']}],
+		]);
+
+		battle.onEvent('Accuracy', battle.format, function (accuracy) {
+			assert.equal(accuracy, 93, 'Wide Lens Fire Blast should be 94% accurate');
+		});
+
+		battle.makeChoices();
+	});
+
+	it(`should chain modifiers in order of the Pokemon's raw speed`, function () {
+		battle = common.createBattle({gameType: 'doubles'}, [[
+			{species: 'Mewtwo', ability: 'victorystar', moves: ['gravity', 'sleeptalk', 'sandattack']},
+			{species: 'Charizard', ability: 'compoundeyes', moves: ['sleeptalk', 'fireblast']},
+		], [
+			{species: 'Bonsly', ability: 'tangledfeet', moves: ['doubleteam', 'sleeptalk']},
+			{species: 'Pyukumuku', ability: 'noguard', moves: ['confuseray', 'sandattack', 'sleeptalk']},
+		]]);
+
+		battle.makeChoices('move sandattack -2, move sleeptalk', 'move doubleteam, move sandattack 2');
+		battle.makeChoices('auto', 'move sleeptalk, move confuseray -1');
+
+		battle.onEvent('Accuracy', battle.format, function (accuracy, target, source, move) {
+			if (move.id !== 'fireblast') return;
+			assert.equal(accuracy, 52);
```

Comment:
My calculations:
Base accuracy: 85%
Chained modifiers: Gravity, Victory Star, Compound Eyes, Tangled Feet: 4892 / 4096
Modified accuracy: 101.52 rounded to 102%
Boosts halve the accuracy to 51%

Of the possible permutations of modifiers, the following chained modifiers are possible:
- 4893 from Tangled Feet/Compound Eyes, Victory Star, Gravity
- 4892 from Tangled Feet/Compound Eyes, Gravity, Victory Star
- 4892 from Gravity/Victory Star, Tangled Feet/Compound Eyes
- All other orderings produce 4891 which results in an overall accuracy of 50%.

Reply:
Oops that's right, I got so caught up on the infinity I forgot to use the right number! I fixed.